lib/db.js

```diff
@@ -118,6 +118,39 @@ class IndexTransaction {
         this.push({ type: 'del', key });
     }
 
+    /**
+     * Adds a condition for the transaction
+     *
+     * @argument {object} condition an object with the following attributes:
+     *                    {
+     *                      notExists: the object key
+     *                    }
+     *
+     * @throws {Error} an error described by the following properties
+     *                 - missingCondition if the condition is empty
+     *                 - pushOnCommittedTransaction if already committed
+     *                 - unsupportedConditionalOperation if the condition is not 'notExists'
+     *                 - missingKey if the key is missing from the op
+     *
+     * @returns {undefined}
+     */
+    addCondition(condition) {
+        if (this.closed) {
+            throw propError('pushOnCommittedTransaction',
+                            'can not add ops to already committed transaction');
+        }
+        if (condition === undefined || Object.keys(condition).length === 0) {
+            throw propError('missingCondition', 'missing condition for conditional put');
+        }
+        if (!('notExists' in condition)) {
+            throw propError('unsupportedConditionalOperation', 'unsupported conditional operation');
+        }
+        if (condition.notExists === undefined) {
+            throw propError('missingKey', 'missing key for conditional operation');
+        }
+        this.operations.push(condition);
```

Comment:
Let's create another object `this.conditions` for conditions: since we are committing a batch, it does not make sense to keep the order of condition checks, as we will always:

1. check all conditions
2. apply the batch (if all conditions are ok)

It will also simplify the check loop (no need to do `splice` etc)